src/api/createDeviceReferenceDataASActions/src/v1/steps.py

```diff
@@ -0,0 +1,142 @@
+from http import HTTPStatus
+
+from domain.api.common_steps.general import parse_event_body
+from domain.api.common_steps.read_product import (
+    parse_path_params,
+    read_product,
+    read_product_team,
+)
+from domain.core.cpm_product.v1 import CpmProduct
+from domain.core.device_reference_data.v1 import DeviceReferenceData
+from domain.core.error import ConfigurationError
+from domain.core.product_key.v1 import ProductKeyType
+from domain.core.questionnaire.v3 import Questionnaire, QuestionnaireResponse
+from domain.repository.device_reference_data_repository.v1 import (
+    DeviceReferenceDataRepository,
+)
+from domain.repository.errors import AlreadyExistsError
+from domain.repository.questionnaire_repository.v2 import QuestionnaireRepository
+from domain.repository.questionnaire_repository.v2.questionnaires import (
+    QuestionnaireInstance,
+)
+from domain.request_models.v1 import (
+    CreateDeviceReferenceAdditionalInteractionsDataParams,
+)
+from domain.response.validation_errors import mark_validation_errors_as_inbound
+
+DEVICE_NAME_MARKER = "AS Additional Interactions"
+
+
+@mark_validation_errors_as_inbound
+def parse_device_reference_data_for_epr_payload(
+    data, cache
+) -> CreateDeviceReferenceAdditionalInteractionsDataParams:
+    payload: dict = data[parse_event_body]
+    return CreateDeviceReferenceAdditionalInteractionsDataParams(**payload)
+
+
+def get_party_key(data, cache) -> str:
+    product: CpmProduct = data[read_product]
+    party_keys = [
+        key.key_value
+        for key in product.keys
+        if key.key_type is ProductKeyType.PARTY_KEY
+    ]
+    try:
+        (party_key,) = party_keys
+    except ValueError:
+        raise ConfigurationError(
+            "Cannot create Additional Interactions in Product without exactly one Party Key"
```

Comment:
Do we want to include not epr product in this error? Or are we moving away from epr language? 

E.g. "Cannot create Additional Interactions for Product without exactly one Party Key (not per product)"